Firebolt API Repository

jlacivita opened the pull request:
# Firebolt API Repository
This repository includes all of the OpenRPC documents from `firebolt-core-sdk`, `firebolt-manage-sdk` and `firebolt-discovery-sdk`.

It is set up using npm workspaces as follows:

```
firebolt-apis/
 - package.json
 - src/
   - descriptions/
   - openrpc/
   - schemas/
   - sdks/
     - core/
       - package.json
       - sdk.config.json
       - test/
       - build/
       - dist/
     - discovery/
       - package.json
       - sdk.config.json
       - test/
       - build/
       - dist/
     - manage/
       - package.json
       - sdk.config.json
       - test/
       - build/
       - dist/     
```

Development can be performed in any SDK individually, but attempting to publish a single SDK will always trigger a rebuild of them all, and ensure all tests pass and version numbers match.

Now that the SDKs are combined in a single repository, the `src/schemas` directory is sort of redundant to the `firebolt-schemas` project. However, I think it does make sense to keep that separate, so that other projects who don't care about the client APIs can access things like Intents, etc.

## Root Package scripts

`npm run compile` calls the `openrpc` task from firebolt-openrpc, and merges/fireboltizes the `src/openrpc` directory..
`npm run slice` clls the `npm run slice` script on every workspace.
`npm run sdks` calls the `npm run sdk` script on every workspace.
`npm run declarations` calls the `npm run declarations` script on every workspace.
`npm run docs` calls the `npm run docs` scripton every workspace.
`npm run validate` calls the `npm run validate` script on every workspace.
`npm run test` calls the `npm run test` script on every workspace.

## Workspace Package scripts

`npm run slice` slices the current SDK's OpenRPC out of the main OpenRPC document and outputs it to ./dist/xxx.json
`npm run sdk` builds the current SDK into the ./build/ directory of it's workspace
`npm run declarations` builds the current SDK's declarations into the ./build/ directory of it's workspace
`npm run docs` builds the current SDK's documentation into the ./build/ directory of it's workspace
`npm run validate` validates the sliced OpenRPC document for this SDK
`npm run test` runs the SDK test suite in this workspace

jlacivita committed:
chore(clean): Removing old bad commit logs

jlacivita committed:
chore(release): Clean up branch/channel mapping

jlacivita committed:
Merge branch 'next' into one-repository

jlacivita committed:
fix: Initiate semantic-release

kpears201 commented on src/openrpc/appcatalog.json:
```diff
@@ -0,0 +1,369 @@
+{
+  "openrpc": "1.2.4",
+  "info": {
+    "title": "AppCatalog",
```
I wonder if we still want this API, since discovery-sdk was never released. Maybe we need to revisit this before we add it here.

jlacivita replied:
Roger, I'll drop it.

Want me to drop the entire `/discovery/` directory, or just this module?

kpears201 commented on src/openrpc/secureStorage.json:
Might be a good opportunity to standardize on snake_case instead of camelCase for the file names.

jlacivita replied:
sure

kpears201 commented on src/openrpc/closedcaptions.json:
maybe call this closed_captions just to standardize on the file naming conventions

kpears201 commented on src/openrpc/secondscreen.json:
second_screen.json

kpears201 commented on src/openrpc/voiceguidance.json:
voice_guidance.json

jlacivita replied:
ditto

kpears201 commented on src/sdks/core/sdk.config.json:
```diff
@@ -0,0 +1,138 @@
+    "info": {
+        "title": "Firebolt Core SDK"
+    },
+    "methods": [
```
wonder if methods is the right name still?

jlacivita replied:
yeah, i was trying to keep it "inspired" by OpenRPC, since lots of the stuff is copied from here directly into the output OpenRPC doc.

It's basically a list of method "queries" so I'd be down to call it `method-queries`

jlacivita replied:
I say keep it as `methods` unless you want to discuss it further.

kpears201 commented on src/sdks/discovery/package.json:
```diff
@@ -0,0 +1,48 @@
+		"wiki": "npx firebolt-openrpc docs --input ./dist/firebolt-discovery-open-rpc.json --output build/docs/markdown",
+		"dist:notest": "npm run clean && npm run slice && npm run validate && npm run sdk && npm run docs && npm run dist:copy && echo 'Firebolt Discovery SDK /dist/ is ready.\n'",
+		"dist:copy": "npm run dist:copy:sdk && npm run dist:copy:docs",
+		"dist:copy:sdk": "mkdirp ./dist && cp -R build/javascript/src dist/lib && cp -r ./dist ../../../dist/discovery && mv ../../../dist/discovery/firebolt-discovery-open-rpc.json ../../../dist/firebolt-discovery-open-rpc.json",
```
wonder if these mv paths will work on Windows

jlacivita replied:
from what I understand, it's basically `mkdir -p` that breaks windows... who is our resident windows user to check for us?

jlacivita replied:
@kevinshahfws : could you help us out here?

jlacivita commented on src/openrpc/systemmessage.json:
```diff
@@ -0,0 +1,106 @@
+{
+    "openrpc": "1.2.4",
+    "info": {
+        "title": "SystemMessage",
```
Roger.

jlacivita committed:
chore(package): Adding directories to workspaces package.jsons

jlacivita committed:
chore(filenames): Using snake_case filenames for modules

kevinshahfws committed:
Updated dryRun mode to true

jlacivita committed:
chore(release): Set up npm publish for each SDK

jlacivita committed:
chore(release): Mapping branch to channel

jlacivita committed:
chore(release): Don't rebuild entire repository for each SDK

jlacivita committed:
chore(release): Adding npmignore to each SDK

jlacivita committed:
chore(release): Update release action step name

jlacivita committed:
chore(release): Return 'latest' for non-prerelease branches

jlacivita committed:
chore(release): Scrub pre-release notes from CHANGELOG.md

jlacivita committed:
chore(release): Cleaning up code

jlacivita committed:
fix(rpc): Better rpc.discover capability

kevinshahfws committed:
Added second example to APIs tagged with property

kevinshahfws committed:
Updated example name to default

kevinshahfws committed:
Corrected account:token capability configuration

kevinshahfws committed:
Removed token:session from config.json

kevinshahfws committed:
Merge pull request #109 from kevinshahfws/openrpc-example-insertion

fix: Added second example to APIs tagged with property

jlacivita committed:
chore(release): Turn off dry-run

semantic-release-bot committed:
chore(release): 0.11.0-one-repository.1 [skip ci]

# [0.11.0-one-repository.1](https://github.com/rdkcentral/firebolt-core-sdk/compare/v0.10.0...v0.11.0-one-repository.1) (2023-04-17)

### Bug Fixes

* Build SDK before npm publish ([#64](https://github.com/rdkcentral/firebolt-core-sdk/issues/64)) ([1994587](https://github.com/rdkcentral/firebolt-core-sdk/commit/199458742f13d46c8d55b60e75112f8de393a8c4))
* Change doc examples to https ([#79](https://github.com/rdkcentral/firebolt-core-sdk/issues/79)) ([0da43e6](https://github.com/rdkcentral/firebolt-core-sdk/commit/0da43e666842bb75d20f9585c0d1f3f0236388ee))
* Don't ignore source files when pulbishing to npm [skip ci] ([19d695b](https://github.com/rdkcentral/firebolt-core-sdk/commit/19d695bcee81a1c7e89749b04d427ec50581c5a1))
* FIRESDK-31 returned response from Lifecycle.ready method ([#81](https://github.com/rdkcentral/firebolt-core-sdk/issues/81)) ([7d13347](https://github.com/rdkcentral/firebolt-core-sdk/commit/7d133473829c30227246b53d32b92c609398344f))
* Initiate semantic-release ([f2fe209](https://github.com/rdkcentral/firebolt-core-sdk/commit/f2fe209ef08a3ce0cb470666da29720e7ab24946))
* **rpc:** Better rpc.discover capability ([91bc54b](https://github.com/rdkcentral/firebolt-core-sdk/commit/91bc54b470dffe14464d95f318d6fb49474ed3ff))
* Updated secure storage spec to allow null as result value ([#84](https://github.com/rdkcentral/firebolt-core-sdk/issues/84)) ([7ed1fed](https://github.com/rdkcentral/firebolt-core-sdk/commit/7ed1fed9ab9f5d84927c4e31c2816e15da0d6f44))
* Word publish scripts ([#78](https://github.com/rdkcentral/firebolt-core-sdk/issues/78)) ([df66ebf](https://github.com/rdkcentral/firebolt-core-sdk/commit/df66ebf642de26dced139998b2834fc8f801e88b))
* Word publish scripts ([#78](https://github.com/rdkcentral/firebolt-core-sdk/issues/78)) ([3ee86bf](https://github.com/rdkcentral/firebolt-core-sdk/commit/3ee86bf587e94d51d662c24d4eb0fe63a799f213))

### Features

* Add Secure Storage module ([#77](https://github.com/rdkcentral/firebolt-core-sdk/issues/77)) ([e633428](https://github.com/rdkcentral/firebolt-core-sdk/commit/e633428843d802b330d5fc6d93a4a5e3e5370521))
* Trigger ([9792bee](https://github.com/rdkcentral/firebolt-core-sdk/commit/9792beee52a1937a0e90bebcd5a80265e5f7eaac))
* Trigger ([01156ca](https://github.com/rdkcentral/firebolt-core-sdk/commit/01156ca1ae023d944984637dad3bb9e5151941c1))
* Trigger ([35cb430](https://github.com/rdkcentral/firebolt-core-sdk/commit/35cb430f7ffb5ece384b8799707aeb945e72aa9e))
* Trigger ([eba005f](https://github.com/rdkcentral/firebolt-core-sdk/commit/eba005f3565475fb01a0d73f907839c621b74e43))
* Trigger ([52bf827](https://github.com/rdkcentral/firebolt-core-sdk/commit/52bf827508bcec2995ccbbae6b614a8b839132bc))
* Trigger ([2a2b739](https://github.com/rdkcentral/firebolt-core-sdk/commit/2a2b73910b96017afa4ee6e3c3d3420591ca61ac))
* Trigger ([ccfc364](https://github.com/rdkcentral/firebolt-core-sdk/commit/ccfc364b342947ab2e1ffadbe5e3825a207c4d83))
* Trigger ([b3e6f83](https://github.com/rdkcentral/firebolt-core-sdk/commit/b3e6f83fd6326b1ddd341ef30baa3c5b63382f62))
* Turn on dryRun ([228e7d9](https://github.com/rdkcentral/firebolt-core-sdk/commit/228e7d97af909933ed2b3a5984d76e957cc98f7f))
* Turn on dryRun ([f90d475](https://github.com/rdkcentral/firebolt-core-sdk/commit/f90d47595def476cd09f4e646f1f99a22180791f))

jlacivita committed:
chore(release): Fix release channel selection

jlacivita committed:
Merge branch 'one-repository' of github.com:rdkcentral/firebolt-core-sdk into one-repository

jlacivita committed:
fix(README): Update Readme

semantic-release-bot committed:
chore(release): 0.11.0-one-repository.2 [skip ci]

# [0.11.0-one-repository.2](https://github.com/rdkcentral/firebolt-core-sdk/compare/v0.11.0-one-repository.1...v0.11.0-one-repository.2) (2023-04-17)

### Bug Fixes

* **README:** Update Readme ([0b8c20a](https://github.com/rdkcentral/firebolt-core-sdk/commit/0b8c20a1f25824b6647a553a1c05a6a8cfac47e8))

jlacivita committed:
fix(release): Stop pushing mono-artifact to npm

jlacivita committed:
Merge branch 'one-repository' of github.com:rdkcentral/firebolt-core-sdk into one-repository

semantic-release-bot committed:
chore(release): 0.11.0-one-repository.3 [skip ci]

# [0.11.0-one-repository.3](https://github.com/rdkcentral/firebolt-core-sdk/compare/v0.11.0-one-repository.2...v0.11.0-one-repository.3) (2023-04-17)

### Bug Fixes

* **release:** Stop pushing mono-artifact to npm ([a7c772e](https://github.com/rdkcentral/firebolt-core-sdk/commit/a7c772e143662f8b7650280eea12262e70bc3f2a))

jlacivita committed:
fix(READMEs): Add various READMEs, remove Discovery SDK

jlacivita committed:
Merge branch 'one-repository' of github.com:rdkcentral/firebolt-core-sdk into one-repository

jlacivita committed:
feat(specification): Add script to generate Firebolt Version Specification JSON

semantic-release-bot committed:
chore(release): 0.11.0-one-repository.4 [skip ci]

# [0.11.0-one-repository.4](https://github.com/rdkcentral/firebolt-core-sdk/compare/v0.11.0-one-repository.3...v0.11.0-one-repository.4) (2023-04-17)

### Bug Fixes

* **READMEs:** Add various READMEs, remove Discovery SDK ([80365af](https://github.com/rdkcentral/firebolt-core-sdk/commit/80365afad63faf3a67a8f679047c3fc618aaa413))

### Features

* **specification:** Add script to generate Firebolt Version Specification JSON ([d417ac4](https://github.com/rdkcentral/firebolt-core-sdk/commit/d417ac4b7f96b314476401d20bf2e2b71bce5de5))

jlacivita committed:
fix(release): Bring back npm plugin, turn of publish

semantic-release-bot committed:
chore(release): 0.11.0-one-repository.5 [skip ci]

# [0.11.0-one-repository.5](https://github.com/rdkcentral/firebolt-core-sdk/compare/v0.11.0-one-repository.4...v0.11.0-one-repository.5) (2023-04-17)

### Bug Fixes

* **release:** Bring back npm plugin, turn of publish ([2fc9196](https://github.com/rdkcentral/firebolt-core-sdk/commit/2fc91962d85e050350415dd628d6144e8569299d))

jlacivita committed:
fix(release): Initiating semantic-release test

jlacivita committed:
fix(release): Only scrub pre-release notes on prod release cycle

also scrubbed version values from subpackages (they're done by CI now)

semantic-release-bot committed:
chore(release): 0.11.0-one-repository.6 [skip ci]

# [0.11.0-one-repository.6](https://github.com/rdkcentral/firebolt-core-sdk/compare/v0.11.0-one-repository.5...v0.11.0-one-repository.6) (2023-04-17)

### Bug Fixes

* **release:** Initiating semantic-release test ([17063e8](https://github.com/rdkcentral/firebolt-core-sdk/commit/17063e885215701dbab982ab058d916ec54badba))
* **release:** Only scrub pre-release notes on prod release cycle ([4cf4de3](https://github.com/rdkcentral/firebolt-core-sdk/commit/4cf4de3bc2df1669191f595190a8fb5d8f955394))

jlacivita committed:
fix(release): Copy broilerplate docs into each workspace

also scrubbed version values from subpackages (they're done by CI now)

jlacivita committed:
Merge branch 'one-repository' of github.com:rdkcentral/firebolt-core-sdk into one-repository

semantic-release-bot committed:
chore(release): 0.11.0-one-repository.7 [skip ci]

# [0.11.0-one-repository.7](https://github.com/rdkcentral/firebolt-core-sdk/compare/v0.11.0-one-repository.6...v0.11.0-one-repository.7) (2023-04-17)

### Bug Fixes

* **release:** Copy broilerplate docs into each workspace ([2e6bd34](https://github.com/rdkcentral/firebolt-core-sdk/commit/2e6bd34b4317e9663aeafc173f6c6f9a6f249e83))

jlacivita committed:
fix(release): Remove redundant build, fix broilerplate copy

jlacivita committed:
Merge branch 'one-repository' of github.com:rdkcentral/firebolt-core-sdk into one-repository

semantic-release-bot committed:
chore(release): 0.11.0-one-repository.8 [skip ci]

# [0.11.0-one-repository.8](https://github.com/rdkcentral/firebolt-core-sdk/compare/v0.11.0-one-repository.7...v0.11.0-one-repository.8) (2023-04-17)

### Bug Fixes

* **release:** Remove redundant build, fix broilerplate copy ([836b246](https://github.com/rdkcentral/firebolt-core-sdk/commit/836b246040d6b92131156b24cc69c401de5d7424))

jlacivita committed:
fix(release): Drop redundant copy of each SDK's /dist/

jlacivita committed:
Merge branch 'one-repository' of github.com:rdkcentral/firebolt-core-sdk into one-repository

jlacivita committed:
fix(release): Fixed untested mv/cp

semantic-release-bot committed:
chore(release): 0.11.0-one-repository.9 [skip ci]

# [0.11.0-one-repository.9](https://github.com/rdkcentral/firebolt-core-sdk/compare/v0.11.0-one-repository.8...v0.11.0-one-repository.9) (2023-04-17)

### Bug Fixes

* **release:** Drop redundant copy of each SDK's /dist/ ([97dd3da](https://github.com/rdkcentral/firebolt-core-sdk/commit/97dd3dadfa278c76a4a99357c9eb3ede6dfc59f7))
* **release:** Fixed untested mv/cp ([a329280](https://github.com/rdkcentral/firebolt-core-sdk/commit/a329280cf099ada6d42e9b3d451a10c35bb60c67))

jlacivita committed:
fix(release): Pointing to next branch for `firebolt-openrpc`

jlacivita committed:
Merge branch 'one-repository' of github.com:rdkcentral/firebolt-core-sdk into one-repository

semantic-release-bot committed:
chore(release): 0.11.0-one-repository.10 [skip ci]

# [0.11.0-one-repository.10](https://github.com/rdkcentral/firebolt-core-sdk/compare/v0.11.0-one-repository.9...v0.11.0-one-repository.10) (2023-04-17)

### Bug Fixes

* **release:** Pointing to next branch for `firebolt-openrpc` ([e25b57a](https://github.com/rdkcentral/firebolt-core-sdk/commit/e25b57ac6986187752e39d78881c0ad529b39226))

kevinshahfws commented:
:tada: This PR is included in version 0.11.0-next.1 :tada:

The release is available on [GitHub release](https://github.com/rdkcentral/firebolt-core-sdk/releases/tag/v0.11.0-next.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

kevinshahfws commented:
:tada: This PR is included in version 0.11.0 :tada:

The release is available on [GitHub release](https://github.com/rdkcentral/firebolt-core-sdk/releases/tag/v0.11.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
